pkg/runtime/reconciler.go

```diff
@@ -505,7 +510,10 @@ func (r *resourceReconciler) deleteResource(
 		return latest, err
 	}
 	rlog.Info("deleted resource")
-	return latest, nil
+
+	// Since the finalizer has been removed, the controller should successfully
```

Comment:
Please update this comment to say exactly what is happening here.
"Return 'nil' resource so that patchStatus attempt is not made. Underlying K8s resource gets deleted after removing finalizer string and patchStatus call will return NotFound error."

Reply:
done!